### PR TITLE
Backport Fix usage of registry for 2.1.0+ from pr #3665

### PR DESCRIPTION
**Pull request by bmac**

### CODE REVIEW

File: packages/ember-data/lib/system/store.js [hunk 1]
```diff
@@ -1742,8 +1742,11 @@ Store = Service.extend({
 
   _modelForMixin: function(modelName) {
     var normalizedModelName = normalizeModelName(modelName);
-    var registry = this.container._registry ? this.container._registry : this.container;
-    var mixin = registry.resolve('mixin:' + normalizedModelName);
+    // container.registry = 2.1
+    // container._registry = 1.11 - 2.0
+    // container = < 1.11
+    var registry = this.container.registry || this.container._registry || this.container;
+    var mixin = this.container.lookupFactory('mixin:' + normalizedModelName);
```
Comment (bmac) [inline packages/ember-data/lib/system/store.js]: Need to double check that `container.lookupFactory` has existed since 1.8 before this can be merged.

Comment (sly7-7) [inline packages/ember-data/lib/system/store.js]: Does https://github.com/emberjs/ember.js/blob/v1.8.0/packages/container/lib/container.js#L322 is enough to confirm ?

Comment (bmac) [inline packages/ember-data/lib/system/store.js]: yes. Thanks @sly7-7

**Comment by fivetanley** [comment]
:+1:

**Pull request merged**